Inc/dec segmentation fault and compile error in Ariane project

Dear Verilator Team, I am working on an Ariane project and using a verilator tool. However, for now, I am struggling with `make verilate`. Following is the description of the problem.

I have installed riscv-gnu-toolchain, riscv-tool with the latest versions available, updated Ariane submodules recursively, installed Verilator with the latest (4.035) version globally from the git master branch, unset $VERILATOR_ROOT by every built, have passed Verilator test examples successfully, downloaded some packages that advised by Verilator page that can help to improve performance of the builds, and so far it seems that I have done everything right following manuals of both [Ariane](https://github.com/pulp-platform/ariane) and [Verilator](https://github.com/verilator/verilator). Also, I am not using SystemC, so I did not download that package, if it may be the case. However, for my Ubuntu 18.04 OS, I can't `make verilate` with `Segmentation fault (core dumped) %Error: Command Failed` error. There are plenty of warnings on pin assignments, delays and expected values which you may see in the attached terminal log file.

Could you help me with the solution of this problem? Initially, I thought that it could be because of my RAM size (6GB) being small, however, it seems it is not the case.
I have googled around old issues, but for now, I did not find anything useful.
Hope to hear back from you soon.

[make_verilate_log.txt](https://github.com/verilator/verilator/files/4708136/make_verilate_log.txt)

@gezalore  thank you for the quick answer. 
After doing that on the $(verilate_command)   line in the make file, line 403 , which now looks like this : $(verilate_command) --debug --gdbbt 
 I am now  back to seeing the original error message  : 

Makefile:403: recipe for target 'verilate' failed
make: *** [verilate] Error 134 

Is the new release happening soon or in a couple of months ?

The problem was related to the merge to fix #2223, this is fixed.  However when I compile and have warnings turned on I get:

```
Variane_testharness__8.cpp: In static member function 'static void Variane_testharness::_combo__TOP__32(Variane_testharness__Syms*)':
Variane_testharness__8.cpp:11691:147: error: expected ';' before '}' token
11691 |             vlTOPp->ariane_testharness__DOT__i_ariane__DOT__csr_regfile_i__DOT__csr_update__DOT__unnamedblk1__DOT__unnamedblk1__DOT____Vincrement1}
      |                                                                                                                                                   ^
      |                                                                                                                                                   ;
Variane_testharness__8.cpp:11691:21: error: statement has no effect [-Werror=unused-value]
```

This is also coming through from the #2223 changes.  @tgorochowik can you please take a look & also get a snippit from whatever is causing this into test_regress?  Thanks.

--- a/src/V3LinkInc.cpp
+++ b/src/V3LinkInc.cpp
@@ -108,6 +108,14 @@ private:
         // Done the loop
         m_insStmtp = NULL;  // Next thing should be new statement
     }
+    virtual void visit(AstNodeIf* nodep) VL_OVERRIDE {
+        m_insStmtp = nodep;
+        iterateAndNextNull(nodep->condp());
+        m_insStmtp = NULL;
+        iterateAndNextNull(nodep->ifsp());
+        iterateAndNextNull(nodep->elsesp());
+        m_insStmtp = NULL;
+    }
     virtual void visit(AstNodeFor* nodep) VL_OVERRIDE {  // LCOV_EXCL_LINE
         nodep->v3fatalSrc(
             "For statements should have been converted to while statements in V3Begin.cpp");
@@ -121,14 +129,6 @@ private:
         m_insStmtp = nodep;
         iterateChildren(nodep);
         m_insStmtp = NULL;  // Next thing should be new statement
-    }
-    virtual void visit(AstNodeBlock* nodep) VL_OVERRIDE {
-        AstNode* insStmtp_prev = m_insStmtp;
-        {
-            m_insStmtp = NULL;
-            iterateChildren(nodep);
-        }
-        m_insStmtp = insStmtp_prev;
     }
     void unsupported_visit(AstNode* nodep) {
         m_unsupportedHere = true;

--- a/test_regress/t/t_array_index_increment.v
+++ b/test_regress/t/t_array_index_increment.v
@@ -112,7 +112,10 @@ module t (/*AUTOARG*/
       if (1 == 1) begin
          pos++;
       end
+      if (pos != 1) $stop;
 
+      pos = 0;
+      if (1 == 1) pos++;
       if (pos != 1) $stop;
 
       $write("*-* All Finished *-*\n");

